### PR TITLE
[RFC][DTypes] Embedding

### PR DESCRIPTION
**Pull request by Godricly**
embedding for DType
need mshadow pull [#136](https://github.com/dmlc/mshadow/pull/136) to be merged

### CODE REVIEW

File: src/operator/embedding.cc [hunk 1]
```diff
@@ -9,12 +9,22 @@
...
+    op = new EmbeddingOp<cpu, DType>(param);
+  });
+  return op;
 }
 
-Operator* EmbeddingProp::CreateOperator(Context ctx) const {
-  DO_BIND_DISPATCH(CreateOp, param_);
+// DO_BIND_DISPATCH comes from operator_common.h
+Operator *EmbeddingProp::CreateOperatorEx(Context ctx, std::vector<TShape> *in_shape,
+                                     std::vector<int> *in_type) const {
```
Comment (vchuravy) [inline src/operator/embedding.cc]: Can you fix this indendation

File: src/operator/embedding.cu [hunk 2]
```diff
@@ -9,8 +9,12 @@
 namespace mxnet {
 namespace op {
 template<>
-Operator* CreateOp<gpu>(EmbeddingParam param) {
-  return new EmbeddingOp<gpu>(param);
+Operator* CreateOp<gpu>(EmbeddingParam param, int dtype) {
+  Operator *op = NULL;
+  MSHADOW_REAL_TYPE_SWITCH(dtype, DType, {
+    op = new EmbeddingOp<gpu, DType>(param);
+  })
```
Comment (vchuravy) [inline src/operator/embedding.cu]: missing semicolon (purely for consistency)

**Comment by vchuravy** [comment]
Looks good, can you squash and rebase?

File: src/operator/embedding.cu (hunk 2 above)
Comment (Godricly) [inline src/operator/embedding.cu]: O,ill added it.

**Comment by Godricly** [comment]
How do I rebase in web?

**Pull request closed without merging**